src/common/datomish/util.cljc

```diff
@@ -101,3 +108,49 @@
 
 (defn mapvals [f m]
   (into (empty m) (map #(vector (first %) (f (second %))) m)))
+
+(defn unblocking-chan?
+  "Returns true if the channel will never block. That is to say, puts
+  into this channel will never cause the buffer to be full."
+  [chan]
+  (a/unblocking-buffer?
+    ;; See http://dev.clojure.org/jira/browse/ASYNC-181.
+    (#?(:cljs .-buf :clj .buf) chan)))
+
+;; Modified from http://dev.clojure.org/jira/browse/ASYNC-23.
+#?(:cljs
+   (deftype UnlimitedBuffer [buf]
+     cljs.core.async.impl.protocols/UnblockingBuffer
+
+     cljs.core.async.impl.protocols/Buffer
+     (full? [this]
+       false)
+     (remove! [this]
+       (.pop buf))
+     (add!* [this itm]
+       (.unshift buf itm))
```

Comment:
Yeah.  There are lots of ways to make this better; a persistent Clojure(Script) deque might even do better than index-munging.  For now, it's not important.